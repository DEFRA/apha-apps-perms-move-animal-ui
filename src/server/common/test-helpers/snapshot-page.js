import { parseDocument } from '../../common/test-helpers/dom.js'
import { createServer } from '~/src/server/index.js'
import { withCsrfProtection } from './csrf.js'
/** @import { Server } from '@hapi/hapi' */

/* global expect, it, beforeAll, afterAll, describe */

/**
 * @param {string} describeMessage
 * @param {string} shouldMessage
 * @param {string} urlPath
 */
export const describePageSnapshot = (
  describeMessage,
  shouldMessage,
  urlPath
) => {
  describe(describeMessage, () => {
    /** @type {Server} */
    let server

    beforeAll(async () => {
      server = await createServer()
      await server.initialize()
    })

    afterAll(async () => {
      await server.stop({ timeout: 0 })
    })

    it(shouldMessage, async () => {
      const { payload } = await server.inject(
        withCsrfProtection({
          method: 'GET',
          url: urlPath
        })
      )

      const content =
        parseDocument(payload).querySelector('#main-content')?.innerHTML

      expect(content).toMatchSnapshot()
    })
  })
}
