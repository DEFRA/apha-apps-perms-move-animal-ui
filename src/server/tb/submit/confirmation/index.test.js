import { describePageSnapshot } from '~/src/server/common/test-helpers/snapshot-page.js'
import { tbConfirmationPage } from './index.js'

const pageTitle = 'Your animal disease movement licence application'

describe('TbConfirmationPage', () => {
  const page = tbConfirmationPage

  it('should have the right key', () => {
    expect(page.key).toBe('tb-submit-confirmation')
  })

  it('should have the right url', () => {
    expect(page.urlPath).toBe('/tb/submit/confirmation')
  })

  it('should have the right page title', () => {
    expect(page.pageTitle).toBe(pageTitle)
  })

  it('should have the right heading', () => {
    expect(page.heading).toBe(pageTitle)
  })

  describePageSnapshot({
    describes: 'TbConfirmationPage',
    it: 'should have the expected content',
    pageUrl: page.urlPath,
    rawState: {
      'tb-applicationReference': 'TB-EXAM-PLE!'
    }
  })
})
