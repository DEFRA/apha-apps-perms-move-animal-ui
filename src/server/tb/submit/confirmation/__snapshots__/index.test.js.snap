// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TbConfirmationPage TbConfirmationPage should have the expected content 1`] = `undefined`;
